### ISSUE TITLE
Zypper repository: change auto-refresh on existing repositories

### ISSUE THREAD

**Issue by mvhirsch** [ansible/ansible-modules-extras#1633]
##### Summary:

I wanted to update an pre-existing repository with `zypper_repository` to set the autorefresh feature to on/off (depending on the repository).
##### Steps To Reproduce:

All four repositories do currently exist after a fresh install, all of them has "refresh => no".

```
$> zypper repos --uri
# | Alias                 | Name                  | Enabled | GPG Check | Refresh | URI                                                              
--+-----------------------+-----------------------+---------+-----------+---------+------------------------------------------------------------------
1 | distro-non-oss        | openSUSE Non-OSS      | Yes     | (r ) Yes  | No      | http://download.opensuse.org/distribution/leap/42.1/repo/non-oss/
2 | distro-oss            | openSUSE OSS          | Yes     | (r ) Yes  | No      | http://download.opensuse.org/distribution/leap/42.1/repo/oss/    
3 | distro-update-non-oss | distro-update-non-oss | Yes     | ( p) Yes  | No      | http://download.opensuse.org/update/leap/42.1/non-oss/           
4 | distro-update-oss     | openSUSE OSS Update   | Yes     | (r ) Yes  | No      | http://download.opensuse.org/update/leap/42.1/oss/ 
```

Then I execute the following tasks.

```
- name: auto-refresh on update repositories
  zypper_repository: name={{ item.name }} repo={{ item.repo }} refresh=yes state=present
  with_items:
    - { name: distro-update-non-oss, repo: "http://download.opensuse.org/update/leap/42.1/non-oss/" }
    - { name: distro-update-oss, repo: "http://download.opensuse.org/update/leap/42.1/oss/" }

- name: auto-refresh off on non-update repositories
  zypper_repository: name={{ item.name }} repo={{ item.repo }} refresh=no state=present                                                                                        
  with_items:
    - { name: distro-non-oss, repo: "http://download.opensuse.org/distribution/leap/42.1/non-oss/" }
    - { name: distro-oss, repo: "http://download.opensuse.org/distribution/leap/42.1/oss/" }

```

I always get success (ok=2, changed=0)
##### Expected Results:

My expectation was that the first two repositories do now have "refresh => yes".
##### Actual Results:

The repositories keep on "refresh => no".

**Comment by alxgu** [comment]
The module checks if the repo exists (with the repo url) and exit with the changed=False statement.
So it ignores checking whether gpgcheck or autorefresh is activated/disabled.

**Comment by robinro** [comment]
possible fix  #1515
